SConstruct

```diff
@@ -90,12 +90,17 @@ from packaging.specifiers import SpecifierSet
 from packaging.version import parse as parse_version
 import SCons
 
+from buildutils import (Option, PathOption, BoolOption, EnumOption, Configuration,
+                        logger, remove_directory, remove_file, test_results,
+                        add_RegressionTest, get_command_output, listify, which,
+                        ConfigBuilder, multi_glob, quoted, add_system_include,
+                        checkout_submodule, check_for_python, check_sundials,
+                        config_error, run_preprocessor, make_relative_path_absolute)
```

Comment:
I'm not sure this one is actually better than the `*`. Otherwise, the rest look good to me :smile:

Reply:
Citing from [PEP 8](https://peps.python.org/pep-0008/#imports):

> Wildcard imports (`from <module> import *`) should be avoided, as they make it unclear which names are present in the namespace, confusing both readers and many automated tools. There is one defensible use case for a wildcard import, which is to republish an internal interface as part of a public API (for example, overwriting a pure Python implementation of an interface with the definitions from an optional accelerator module and exactly which definitions will be overwritten isn’t known in advance).

The alternative would be to use `import buildutils as bu`, but I honestly think that the long list at the beginning of the module is preferable.

Reply:
Yes, I'm familiar with PEP 8. I think in this case, it makes more sense to ignore pep 8's recommendation. I should explain my reasoning.

In this case, I think this long block is less readable and maintainable than the *. I feel that way because, well , readability is just vibes as the kids say. But I do think it's harder to maintain this long list here rather than in the dunder all in buildutils. 

There are a couple of other reasons that pep 8 makes that recommendation against * (although they're not stated explicitly), which I think don't apply here.

1. I also considered that the * import can't be resolved by the LSP implementation. However, even the explicit import can't be resolved because `site_scons` is added to the path by SCons at runtime.
2. Similarly, I don't think naming everything on the import makes it more obvious where they come from because buildutils is hidden inside site_scons.
2. I also considered namespace pollution from buildutils. Since we define dunder all in buildutils, we are essentially defining what should be exported rather than imported. In any case, this avoids the pollution issue since we use nearly every exported member in the main SConstruct. 
3. SCons also does a bunch of magical injections of objects at runtime that you can use without importing. So in that sense using a * fits in the same pattern as SCons is already using. 

So, considering all that, my opinion is that the * is the better tradeoff ☺️

Reply:
I think we're approaching the problem with the same facts and just prioritizing differently. We may have to agree to disagree and call in @speth to break the tie 😁

Reply:
To me, this ends up being a pretty minor difference in the end. I think there end up being a few points in favor of importing the specific names, though.

One is that every Python user understands what `from x import y` means, whereas the use of `__all__` is a somewhat obscure detail.

Also, I think the long list, as ugly as it is, is easier to maintain as part of an `import` statement, since in that case your IDE can inform you when there's an unused import, whereas I don't think this is meaningful/possible for `__all__`.